core/src/main/java/io/questdb/cutlass/text/CairoTextWriter.java

```diff
@@ -118,17 +115,32 @@ public long getWrittenLineCount() {
         return writer == null ? 0 : writer.size() - _size;
     }
 
-    public void of(CharSequence name, boolean overwrite, boolean durable, int atomicity) {
+    public void of(CharSequence name, boolean overwrite, boolean durable, int atomicity, int partitionBy, CharSequence timestampIndexCol) {
         this.tableName = name;
         this.overwrite = overwrite;
         this.durable = durable;
         this.atomicity = atomicity;
+        this.partitionBy = partitionBy;
+        this.timestampIndexCol = timestampIndexCol;
     }
 
     @Override
     public void onFields(long line, ObjList<DirectByteCharSequence> values, int valuesLength) {
-        final TableWriter.Row w = writer.newRow();
+        long timestamp = 0L;
+        if (timestampIndex != -1 && types.getQuick(timestampIndex).getType() == ColumnType.TIMESTAMP) {
+            TimestampAdapter adapter = (TimestampAdapter) types.getQuick(timestampIndex);
+            final DirectByteCharSequence dbcs = values.getQuick(timestampIndex);
+            try {
+                timestamp = adapter.getTimestamp(dbcs);
+            } catch (NumericException e) {
+                logError(line, timestampIndex, dbcs);
```

Comment:
the error handling here needs to either:

- When atomicity == SKIP_ALL, throw CairoException. There is no need to rollback row, it hasn't been created
- do the logging but exit method via `return`.

Reply:
I opted for the second.